### PR TITLE
Polyhedron demo: add Depth-peeling and Rendering Helpers

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
This PR implements a Depth-peeling system and adds alpha management for the Scene_surface_mesh_item and the Scene_c3t3_item. 
An interface or a helper for this kind of items (with alpha != 1.0f) should be added later to make it easier to use, and to factorize the menu management.

c3t3_alpha_snapshot

### CODE REVIEW

File: Maintenance/deb/sid/debian/README.source [hunk 1]
```diff
@@ -1,11 +0,0 @@
...
---------------------------
-
-This package uses quilt to manage all modifications to the upstream source.
-Changes are stored in the source package as diffs in debian/patches and
-applied during the build.
-
-See /usr/share/doc/quilt/README.source for a detailed explanation.
-
- -- Joachim Reichel <joachim.reichel@gmx.de>  Sun, 15 Jan 2012 14:38:04 +0100
-
```
Comment (lrineau) [inline Maintenance/deb/sid/debian/README.source]: Why do you remove those files?! They are used by Joachim Reichel to maintain the Debian packages.

Comment (maxGimeno) [reply Maintenance/deb/sid/debian/README.source]: It is automatic. Every time I create a new branch from master it does that.

Comment (maxGimeno) [reply Maintenance/deb/sid/debian/README.source]: It seems to be a bug in a personnal convenience script. I'll look into it and fix it.

**Comment by afabri** [comment]
compiles, runs, and does the right thing on Windows.

File: Polyhedron/demo/Polyhedron/CMakeLists.txt [hunk 2]
```diff
@@ -204,19 +204,19 @@ if(CGAL_Qt5_FOUND AND Qt5_FOUND AND OPENGL_FOUND )
...
   target_link_libraries(point_dialog
-    PUBLIC Qt5::OpenGL Qt5::Gui Qt5::Script Qt5::Widgets)
+    PUBLIC Qt5::OpenGL Qt5::Gui Qt5::Script Qt5::Widgets Qt5::Xml )
 
   macro(add_item item_name)
     add_library(${item_name} SHARED ${ARGN})
     target_link_libraries(${item_name}
       PUBLIC demo_framework ${CGAL_LIBRARIES} ${Boost_LIBRARIES}
-      Qt5::OpenGL Qt5::Gui Qt5::Script Qt5::Widgets)
+      Qt5::OpenGL Qt5::Gui Qt5::Script Qt5::Widgets Qt5::Xml)
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/CMakeLists.txt]: Please revert this commit. It is not right. `Qt5::Xml` is only a dependency of QGLViewer, now integrated in `CGAL::Qt5`.

**Comment by lrineau** [comment]
There are warnings:
```
In file included from /mnt/testsuite/include/CGAL/Three/Scene_item_config.h:28:0,
                 from /mnt/testsuite/include/CGAL/Three/Scene_item.h:27,
                 from /mnt/testsuite/include/CGAL/Three/Scene_group_item.h:29,
                 from /home/cgal_tester/build/src/cmake/platforms/Debian-Stable/test/Polyhedron_Demo/Scene_group_item.cpp:1:
/home/cgal_tester/build/src/cmake/platforms/Debian-Stable/test/Polyhedron_Demo/Scene_group_item.cpp: In member function 'virtual bool CGAL::Three::Scene_group_item::isEmpty() const':
/usr/include/x86_64-linux-gnu/qt5/QtCore/qglobal.h:963:5: warning: this 'for' clause does not guard... [-Wmisleading-indentation]
     for (variable = *_container_.i; _container_.control; _container_.control = 0)
     ^
/home/cgal_tester/build/src/cmake/platforms/Debian-Stable/test/Polyhedron_Demo/Scene_group_item.cpp:25:3: note: in expansion of macro 'Q_FOREACH'
   Q_FOREACH(Scene_interface::Item_id id, children)
   ^
/home/cgal_tester/build/src/cmake/platforms/Debian-Stable/test/Polyhedron_Demo/Scene_group_item.cpp:29:5: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the 'for'
     return true;
     ^~~~~~
In file included from /mnt/testsuite/include/CGAL/Three/Scene_item_config.h:28:0,
                 from /mnt/testsuite/include/CGAL/Three/Scene_item.h:27,
                 from /mnt/testsuite/include/CGAL/Three/Scene_group_item.h:29,
                 from /home/cgal_tester/build/src/cmake/platforms/Debian-Stable/test/Polyhedron_Demo/Scene_group_item.cpp:1:
/home/cgal_tester/build/src/cmake/platforms/Debian-Stable/test/Polyhedron_Demo/Scene_group_item.cpp: In member function 'virtual bool CGAL::Three::Scene_group_item::supportsRenderingMode(RenderingMode) const':
/usr/include/x86_64-linux-gnu/qt5/QtCore/qglobal.h:963:5: warning: this 'for' clause does not guard... [-Wmisleading-indentation]
     for (variable = *_container_.i; _container_.control; _container_.control = 0)
     ^
/home/cgal_tester/build/src/cmake/platforms/Debian-Stable/test/Polyhedron_Demo/Scene_group_item.cpp:40:3: note: in expansion of macro 'Q_FOREACH'
   Q_FOREACH(Scene_interface::Item_id id, children)
   ^
/home/cgal_tester/build/src/cmake/platforms/Debian-Stable/test/Polyhedron_Demo/Scene_group_item.cpp:43:5: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the 'for'
     return true;
     ^~~~~~
```
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.13-Ic-35/Polyhedron_Demo/TestReport_gimeno_Debian-Stable.gz

I have verified and that is true that the indentation is misleading:
https://github.com/CGAL/cgal/blob/a0957926832e06d18ee60aa1a26080dc17815377/Polyhedron/demo/Polyhedron/Scene_group_item.cpp#L24-L45

and the code of `Scene_group_item::isEmpty` is buggy: it already returns `true`!

File: Polyhedron/demo/Polyhedron/Scene_group_item.cpp [hunk 3]
```diff
@@ -24,9 +24,9 @@ bool Scene_group_item::isFinite() const
 bool Scene_group_item::isEmpty() const {
   Q_FOREACH(Scene_interface::Item_id id, children)
     if(!getChild(id)->isEmpty()){
-            return true;
-        }
-    return true;
+      return dalse;
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Scene_group_item.cpp]: Maybe you should try to compile your code!

**Comment by lrineau** [comment]
1. Travis error:
```
In file included from /home/travis/build/CGAL/cgal/dep_check_build/test_headers.cpp:870:
/home/travis/build/CGAL/cgal/Three/include/CGAL/Three/Point_container.h:98:11: error: 
      unknown type name 'Point_d'; did you mean '::Point_d'?
  mutable Point_d* d;
          ^~~~~~~
          ::Point_d
/home/travis/build/CGAL/cgal/Three/include/CGAL/Three/Point_container.h:37:8: note: 
      '::Point_d' declared here
struct Point_d;
       ^
```
https://travis-ci.org/CGAL/cgal/jobs/391225062#L6349

2. There is a conflict with `master`, now.

3. The documentation has errors too:
```
/home/cgal-testsuite/cgal_doc_build/CGAL-4.13-Ic-35/include/CGAL/Three/Primitive_container.h:74: warning: argument 'fbo' of command @param is not found in the argument list of CGAL::Three::Primitive_container::draw(CGAL::Three::Viewer_interface *viewer, bool is_color_uniform)=0
/home/cgal-testsuite/cgal_doc_build/CGAL-4.13-Ic-35/include/CGAL/Three/Scene_item.h:460: warning: explicit link request to 'computeElements()' could not be resolved
/home/cgal-testsuite/cgal_doc_build/CGAL-4.13-Ic-35/include/CGAL/Three/Scene_item_rendering_helper.h:206: warning: explicit link request to 'computeElements()' could not be resolved
/home/cgal-testsuite/cgal_doc_build/CGAL-4.13-Ic-35/include/CGAL/Three/Triangle_container.h:78: warning: argument 'fbo' of command @param is not found in the argument list of CGAL::Three::Triangle_container::draw(CGAL::Three::Viewer_interface *viewer, bool is_color_uniform)
```

**Comment by lrineau** [comment]
@maxGimeno Please fix the conflicts in this branch, by a merge of `master` (not a rebase).

**Comment by lrineau** [comment]
And then of course, test again if the transparency still work.

**Comment by lrineau** [comment]
There are a lot of warnings, in the documentation testsuite, probably because of this PR:
```
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:144: warning: Member setSelected(bool) (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:145: warning: Member setColor(QColor) (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Edge_container.h:95: warning: Member getPlane() const (function) of class CGAL::Three::Edge_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Edge_container.h:96: warning: Member getFrameMatrix() const (function) of class CGAL::Three::Edge_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Edge_container.h:98: warning: Member setPlane(const QVector4D &) (function) of class CGAL::Three::Edge_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Edge_container.h:99: warning: Member setFrameMatrix(const QMatrix4x4 &) (function) of class CGAL::Three::Edge_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:203: warning: Member isSelected() const (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:204: warning: Member getColor() const (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:144: warning: Member setSelected(bool) (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:145: warning: Member setColor(QColor) (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Point_container.h:90: warning: Member getPlane() const (function) of class CGAL::Three::Point_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Point_container.h:91: warning: Member getFrameMatrix() const (function) of class CGAL::Three::Point_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Point_container.h:93: warning: Member setFrameMatrix(const QMatrix4x4 &) (function) of class CGAL::Three::Point_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:203: warning: Member isSelected() const (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:204: warning: Member getColor() const (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:144: warning: Member setSelected(bool) (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:145: warning: Member setColor(QColor) (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:203: warning: Member isSelected() const (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:204: warning: Member getColor() const (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:144: warning: Member setSelected(bool) (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:145: warning: Member setColor(QColor) (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Triangle_container.h:92: warning: Member getShrinkFactor() (function) of class CGAL::Three::Triangle_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Triangle_container.h:93: warning: Member getPlane() (function) of class CGAL::Three::Triangle_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Triangle_container.h:94: warning: Member getAlpha() (function) of class CGAL::Three::Triangle_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Triangle_container.h:96: warning: Member setShrinkFactor(const float &) (function) of class CGAL::Three::Triangle_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Triangle_container.h:97: warning: Member setPlane(const QVector4D &) (function) of class CGAL::Three::Triangle_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Triangle_container.h:98: warning: Member setAlpha(const float &) (function) of class CGAL::Three::Triangle_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:203: warning: Member isSelected() const (function) of class CGAL::Three::Primitive_container is not documented.
.../CGAL-4.13-Ic-47/include/CGAL/Three/Primitive_container.h:204: warning: Member getColor() const (function) of class CGAL::Three::Primitive_container is not documented.
```
https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-4.13-Ic-47/logs_master/Three.log

**Comment by lrineau** [comment]
Successfully tested in [CGAL-4.13-Ic-49][*], and then merged.

  [*]: https://cgal.geometryfactory.com/CGAL/testsuite/results-4.13-Ic-49.shtml

**Pull request merged**